core/app/models/spree/credit_card.rb

```diff
@@ -97,8 +105,10 @@ def set_last_digits
     # @return [String] the credit card type if it can be determined from the
     #   number, otherwise the empty string
     def try_type_from_number
-      numbers = number.delete(' ') if number
-      CARD_TYPES.find{ |type, pattern| return type.to_s if numbers =~ pattern }.to_s
+      CARD_TYPES.each do |type, pattern|
```

Comment:
`.detect` would return us (for example) `['visa', /^4\d{12}(\d{3})?(\d{3})?$/]` so the code would need to be something like

```
card_type_pair = CARD_TYPES.detect { |type, pattern| numbers =~ pattern }
if card_type_pair
  card_type_pair[0]
else
  ''
end
```

I think the return is more clear